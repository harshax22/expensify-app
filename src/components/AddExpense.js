import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm'
import { addExpense } from '../actions/expenses'

const AddExpensePage = (props) => (
    <ExpenseForm 
    onSubmit = {(expense) => {
       props.dispatch(addExpense(expense))
       props.history.push('/')
    }}
    />
);

export default connect()(AddExpensePage);