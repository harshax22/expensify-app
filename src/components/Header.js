import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <div>
    <NavLink to = "/" activeClassName = "is-active" exact = {true}>Visit the Dashboard Page!</NavLink>
    <NavLink to = "/create" activeClassName = "is-active">Create a new Expense!</NavLink>
    
     </div>
    </header>
   
   
);

export default Header;