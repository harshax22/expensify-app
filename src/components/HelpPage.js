import React from 'react';

const EditPage = () => (
    <div>
        This is the Support Page
    </div>
);

export default EditPage;