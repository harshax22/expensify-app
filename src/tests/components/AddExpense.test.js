import React from 'react'
import { shallow } from 'enzyme';
import { AddExpensePage } from '../../components/AddExpense';
import toJson from 'enzyme-to-json';
import expenses from '../fixtures/expenses';

let addExpense, history, wrapper;

beforeEach(() => {
     addExpense = jest.fn();
     history = { push: jest.fn() };
     wrapper = shallow(<AddExpensePage addExpense={addExpense} history={history}/>);
});

test('should render Add Expense Correctly', () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('should set up onSubmit', () => {
wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
expect(history.push).toHaveBeenLastCalledWith('/');
expect(addExpense).toHaveBeenLastCalledWith(expenses[1]);
})