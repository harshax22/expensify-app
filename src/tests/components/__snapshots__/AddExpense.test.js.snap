// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Add Expense Correctly 1`] = `
<ExpenseForm
  onSubmit={[Function]}
/>
`;
